### ISSUE TITLE
Version 1.38 - Unable to see NSCT text with errors at each dmg tick

### ISSUE THREAD

**Issue by hrlmartins** [Justw8/NameplateSCT#41]
Follow up from: https://github.com/Justw8/NameplateSCT/issues/40

Yellow, this version gets me the following error when doing any kind of damage

```lua
35x NameplateSCT/NameplateSCT.lua:925: attempt to compare number with nil
[string "@NameplateSCT/NameplateSCT.lua"]:925: in function `DamageEvent'
[string "@NameplateSCT/NameplateSCT.lua"]:771: in function <NameplateSCT/NameplateSCT.lua:739>
[string "=(tail call)"]: ?
[string "@Bartender4/libs/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:109: in function <...er4/libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:109>
[string "=[C]"]: ?
[string "@Bartender4/libs/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:19: in function <...er4/libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:15>
[string "@Bartender4/libs/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:54: in function `Fire'
[string "@Bartender4/libs/AceEvent-3.0-4/AceEvent-3.0.lua"]:120: in function <Bartender4/libs/AceEvent-3.0/AceEvent-3.0.lua:119>

Locals:
self = <table> {
 modules = <table> {
 }
 defaultModuleLibraries = <table> {
 }
 menu = Frame {
 }
 baseName = "NameplateSCT"
 name = "NameplateSCT"
 orderedModules = <table> {
 }
 db = <table> {
 }
 enabledState = true
 defaultModuleState = true
 frame = Frame {
 }
}
guid = "Creature-0-1461-1-3834-5429-000022C886"
spellName = "Flame Shock"
amount = 3209187
overkill = 3207092
school = 12
crit = false
spellId = 188389
absorbed = nil
amount = 3209187
absorbed = 0
(*temporary) = nil
(*temporary) = 3209187
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to compare number with nil"
playerGUID = "Player-1403-098A4A10"
guidToUnit = <table> {
 Creature-0-1461-1-3834-5429-000022C886 = "nameplate1"
}
NameplateSCT = <table> {
 modules = <table> {
 }
 defaultModuleLibraries = <table> {
 }
 menu = Frame {
 }
 baseName = "NameplateSCT"
 name = "NameplateSCT"
 orderedModules = <table> {
 }
 db = <table> {
 }
 enabledState = true
 defaultModuleState = true
 frame = Frame {
 }
}
L = <table> {
 BlizzardSCT = "BlizzardSCT"
 Auto Attacks = "Auto Attacks"
 Truncate Method:

Western:
  1000=>1K,
  1000K=>1M
Asia East:
  10000=>1w = "Truncate Method:

Western:
  1000=>1K,
  1000K=>1M
Asia East:
  10000=>1w"
 Right = "Right"
 Embiggen Crits = "Embiggen Crits"
 Parried = "Parried"
 Immune = "Immune"
 Display Off-Target Text = "Display Off-Target Text"
 Shake = "Shake"
 Icon Scale = "Icon Scale"
 Let Masuqe skin the icons = "Let Masuqe skin the icons"
 Font Flags = "Font Flags"
 NPCs = "NPCs"
 Hide Small Hits = "Hide Small Hits"
 Bottom = "Bottom"
 Y Variance = "Y Variance"
 Scale down hits that are below a running average of your recent damage output = "Scale down hits that are below a running average of your recent damage output"
 Alpha = "Alpha"
 %s (A: %s) = "%s (A: %s)"
 Personal SCT = "Personal SCT"
 Reflected = "Reflected"
 Default Color = "Default Color"
 Default speed: 1 = "Default speed: 1"
 Show Absorbed Damage shown as: '5 (A: 3)' where A: 3 is the absorbed amount = "Show Absorbed Damage shown as: '5 (A: 3)' where A: 3 is the absorbed amount"
 Bottom Right = "Bottom Right"
 Y Offset = "Y Offset"
 Show Truncated Letter = "Show Truncated Letter"
 Appearance/Offsets = "Appearance/Offsets"
 Display Overkill = "Display Overkill"
 Disabled = "Disabled"
 Y Offset Personal SCT = "Y Offset Personal SCT"
 X Variance = "X Variance"
 High = "High"
 X Offset = "X Offset"
 X Offset Personal SCT = "X Offset Personal SCT"
 Western = "Western"
 Do Not Truncate = "Do Not Truncate"
 Use Crit Color = "Use Crit Color"
 Vertical Up = "Vertical Up"
 Enable Masque = "Enable Masque"
 Resisted = "Resisted"
 Vertical Down = "Vertical Down"
 Size = "Size"
 Rainfall = "Rainfall"
 Use Separate Off-Target Strata = "Use Separate Off-Target Strata"
 Use Damage Type Color = "Use Damage Type Color"
 Truncate Number = "Truncate Number"
 Truncate Letter East Asia = "Truncate Letter East Asia"
 Monochrome = "Monochrome"
 Hide hits that are below this threshold. = "Hide hits that are below this threshold."
 Top = "Top"
 Low = "Low"
 Font = "Font"
 Default = "Default"
 Position = "Position"
 Thick Outline = "Thick Outline"
 Text Shadow = "Text Shadow"
 Has soft max/min, you can type whatever you'd like into the editbox = "Has soft max/min, you can type whatever you'd like into the editbox"
 Bottom Left = "Bottom Left"
 Embiggen Crits Scale = "Embiggen Crits Scale"
 Top Left = "Top Left"
 Off-Target Strata = "Off-Target Strata"
 Off-Target Text Appearance = "Off-Target Text Appearance"
 Enable = "Enable"
 Criticals = "Criticals"
 Left = "Left"
 Top Right = "Top Right"
 Scale Down Small Hits = "Scale Down Small Hits"
 Icon X Offset = "Icon X Offset"
 Spells = "Spells"
 Target Strata = "Target Strata"
 Spellid/Spellname seperated by line

White hits/melee = melee = "Spellid/Spellname seperated by line

White hits/melee 
```

also the following one when I try to open the nsct options with `/nsct`

```lua
2x NameplateSCT/NameplateSCT.lua:1980: attempt to call global 'InterfaceOptionsFrame_OpenToCategory' (a nil value)
[string "@NameplateSCT/NameplateSCT.lua"]:1980: in function `?'
[string "@AstralKeys/Libs/AceConsole-3.0-7/AceConsole-3.0.lua"]:94: in function `?'
[string "@Blizzard_ChatFrameBase/Mainline/ChatFrame.lua"]:5440: in function `ChatEdit_ParseText'
[string "@Blizzard_ChatFrameBase/Mainline/ChatFrame.lua"]:5092: in function `ChatEdit_SendText'
[string "@Blizzard_ChatFrameBase/Mainline/ChatFrame.lua"]:5128: in function `ChatEdit_OnEnterPressed'
[string "*ChatFrame.xml:140_OnEnterPressed"]:1: in function <[string "*ChatFrame.xml:140_OnEnterPressed"]:1>

Locals:
self = <table> {
 modules = <table> {
 }
 defaultModuleLibraries = <table> {
 }
 menu = Frame {
 }
 baseName = "NameplateSCT"
 name = "NameplateSCT"
 orderedModules = <table> {
 }
 db = <table> {
 }
 enabledState = true
 defaultModuleState = true
 frame = Frame {
 }
}
(*temporary) = nil
(*temporary) = Frame {
 obj = <table> {
 }
 name = "NameplateSCT"
}
(*temporary) = "attempt to call global 'InterfaceOptionsFrame_OpenToCategory' (a nil value)"
```

Thank you!!!

**Comment by Justw8** [comment]
I think I Fixed the first error, will look into the slash command.

**Comment by kiruz-0** [comment]
Got similar issue with nil

4x NameplateSCT/NameplateSCT.lua:925: attempt to compare number with nil
[string "@NameplateSCT/NameplateSCT.lua"]:925: in function `DamageEvent'
[string "@NameplateSCT/NameplateSCT.lua"]:771: in function <NameplateSCT/NameplateSCT.lua:739>
[string "=(tail call)"]: ?
[string "@AllTheThings/lib/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:109: in function <...ings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:109>
[string "=[C]"]: ?
[string "@AllTheThings/lib/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:19: in function <...ings/lib/CallbackHandler-1.0/CallbackHandler-1.0.lua:15>
[string "@AllTheThings/lib/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:54: in function `Fire'
[string "@BlizzMove/Libs/AceEvent-3.0-4/AceEvent-3.0.lua"]:120: in function <BlizzMove/Libs/AceEvent-3.0/AceEvent-3.0.lua:119>

Locals:
self = <table> {
 modules = <table> {
 }
 defaultModuleLibraries = <table> {
 }
 menu = Frame {
 }
 baseName = "NameplateSCT"
 name = "NameplateSCT"
 orderedModules = <table> {
 }
 db = <table> {
 }
 enabledState = true
 defaultModuleState = true
 frame = Frame {
 }
}
guid = "Creature-0-4234-2444-9721-193302-000022B1CA"
spellName = "Keg Smash"
amount = 34678
overkill = -1
school = 1
crit = false
spellId = 121253
absorbed = nil
amount = 34678
absorbed = 0
(*temporary) = nil
(*temporary) = 34678
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to compare number with nil"
playerGUID = "Player-3713-0ABE8D5E"
guidToUnit = <table> {
 Creature-0-4234-2444-9721-191404-000022B17F = "nameplate2"
 Creature-0-4234-2444-9721-193302-000022B1CA = "nameplate1"
}
NameplateSCT = <table> {
 modules = <table> {
 }
 defaultModuleLibraries = <table> {
 }
 menu = Frame {
 }
 baseName = "NameplateSCT"
 name = "NameplateSCT"
 orderedModules = <table> {
 }
 db = <table> {
 }
 enabledState = true
 defaultModuleState = true
 frame = Frame {
 }
}
L = <table> {
 BlizzardSCT = "BlizzardSCT"
 Auto Attacks = "Auto Attacks"
 Truncate Method:

Western:
  1000=>1K,
  1000K=>1M
Asia East:
  10000=>1w = "Truncate Method:

Western:
  1000=>1K,
  1000K=>1M
Asia East:
  10000=>1w"
 Right = "Right"
 Embiggen Crits = "Embiggen Crits"
 Parried = "Parried"
 Immune = "Immune"
 Display Off-Target Text = "Display Off-Target Text"
 Shake = "Shake"
 Icon Scale = "Icon Scale"
 Let Masuqe skin the icons = "Let Masuqe skin the icons"
 Font Flags = "Font Flags"
 NPCs = "NPCs"
 Hide Small Hits = "Hide Small Hits"
 Bottom = "Bottom"
 Y Variance = "Y Variance"
 Scale down hits that are below a running average of your recent damage output = "Scale down hits that are below a running average of your recent damage output"
 Alpha = "Alpha"
 %s (A: %s) = "%s (A: %s)"
 Personal SCT = "Personal SCT"
 Reflected = "Reflected"
 Default Color = "Default Color"
 Default speed: 1 = "Default speed: 1"
 Show Absorbed Damage shown as: '5 (A: 3)' where A: 3 is the absorbed amount = "Show Absorbed Damage shown as: '5 (A: 3)' where A: 3 is the absorbed amount"
 Bottom Right = "Bottom Right"
 Y Offset = "Y Offset"
 Show Truncated Letter = "Show Truncated Letter"
 Appearance/Offsets = "Appearance/Offsets"
 Display Overkill = "Display Overkill"
 Disabled = "Disabled"
 Y Offset Personal SCT = "Y Offset Personal SCT"
 X Variance = "X Variance"
 High = "High"
 X Offset = "X Offset"
 X Offset Personal SCT = "X Offset Personal SCT"
 Western = "Western"
 Do Not Truncate = "Do Not Truncate"
 Use Crit Color = "Use Crit Color"
 Vertical Up = "Vertical Up"
 Enable Masque = "Enable Masque"
 Resisted = "Resisted"
 Vertical Down = "Vertical Down"
 Size = "Size"
 Rainfall = "Rainfall"
 Use Separate Off-Target Strata = "Use Separate Off-Target Strata"
 Use Damage Type Color = "Use Damage Type Color"
 Truncate Number = "Truncate Number"
 Truncate Letter East Asia = "Truncate Letter East Asia"
 Monochrome = "Monochrome"
 Hide hits that are below this threshold. = "Hide hits that are below this threshold."
 Top = "Top"
 Low = "Low"
 Font = "Font"
 Default = "Default"
 Position = "Position"
 Thick Outline = "Thick Outline"
 Text Shadow = "Text Shadow"
 Has soft max/min, you can type whatever you'd like into the editbox = "Has soft max/min, you can type whatever you'd like into the editbox"
 Bottom Left = "Bottom Left"
 Embiggen Crits Scale = "Embiggen Crits Scale"
 Top Left = "Top Left"
 Off-Target Strata = "Off-Target Strata"
 Off-Target Text Appearance = "Off-Target Text Appearance"
 Enable = "Enable"
 Criticals = "Criticals"
 Left = "Left"
 Top Right = "Top Right"
 Scale Down Small Hits = "Scale Down Small Hits"
 Icon X Offset = "Icon X Offset"
 Spells = "Spells"
 Target Strata = "Target Strata"
 Spellid/Spellname seperated by line

White hits/melee = melee =

**Comment by Justw8** [comment]
> 4x NameplateSCT/NameplateSCT.lua:925: attempt to compare number with nil
should be fixed with 1.39

**Comment by Justw8** [comment]
Pushed a fix for the slash command not opening the menu. 

https://github.com/Justw8/NameplateSCT/commit/fb231a4801f7ccc37856c98d611922bb4201c0e6

let me know if problems still persist with the latest alpha in about 15 minutes.
